### PR TITLE
#59. ProjectsApi contracts endpoint.

### PR DESCRIPTION
**Pull request by criske**
- implemented and tested.
- added output abstractions for json Contract/Contracts.
- updated puzzle.

PR for #59

### CODE REVIEW

File: src/main/java/com/selfxdsd/selfweb/api/output/JsonContract.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                .add("repoFullName", contract.contractId()
+                    .getRepoFullName())
+                .add("contributorUsername", contract.contractId()
+                    .getContributorUsername())
+                .add("provider", contract.contractId().getProvider())
+                .add("role", contract.contractId().getRole())
+                .build())
+            .add("project", new JsonProject(contract.project()))
+            .add("value", NumberFormat
+                .getCurrencyInstance(Locale.US).format(contract.value()))
```
Comment (amihaiemil) [inline src/main/java/com/selfxdsd/selfweb/api/output/JsonContract.java:33]: @criske We should also add the hourly rate. 
The contract's value is the sum of all ongoing Tasks and current Invoice :D 

We should also transform it from cents to dollars (see how JsonManager does that).

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@criske see comments :D

File: src/main/java/com/selfxdsd/selfweb/api/output/JsonContract.java (hunk 1 above)
Comment (criske) [reply src/main/java/com/selfxdsd/selfweb/api/output/JsonContract.java:33]: @amihaiemil should I format the commission in JsonManager as currency display too?

Comment (amihaiemil) [reply src/main/java/com/selfxdsd/selfweb/api/output/JsonContract.java]: @criske maybe in another PR, but what does this "currency display" actually do? :D

**Comment by amihaiemil** [comment]
@rultor merge it

File: src/main/java/com/selfxdsd/selfweb/api/output/JsonContract.java (hunk 1 above)
Comment (criske) [reply src/main/java/com/selfxdsd/selfweb/api/output/JsonContract.java]: @amihaiemil for example `10` is displayed as `$10.00`

**Pull request merged**